e2e/dir_scan_test.go

```diff
@@ -0,0 +1,184 @@
+			updateScanConfig := UpdateScanConfigToStartNow(apiScanConfig)
+			err = client.PatchScanConfig(ctx, *apiScanConfig.Id, updateScanConfig)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("waiting until scan starts")
+			scanParams := apitypes.GetScansParams{
+				Filter: to.Ptr(fmt.Sprintf(
+					"scanConfig/id eq '%s' and status/state ne '%s' and status/state ne '%s'",
+					*apiScanConfig.Id,
+					apitypes.ScanStatusStateDone,
+					apitypes.ScanStatusStateFailed,
+				)),
+			}
+			var scans *apitypes.Scans
+			gomega.Eventually(func() bool {
+				scans, err = client.GetScans(ctx, scanParams)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				if len(*scans.Items) == 1 {
+					reportFailedConfig.objects = append(
+						reportFailedConfig.objects,
+						APIObject{"scan", fmt.Sprintf("id eq '%s'", *(*scans.Items)[0].Id)},
+					)
+					return true
+				}
+				return false
+			}, DefaultTimeout, time.Second).Should(gomega.BeTrue())
+
+			ginkgo.By("waiting until scan state changes to done")
+			scanParams = apitypes.GetScansParams{
+				Filter: to.Ptr(fmt.Sprintf(
+					"scanConfig/id eq '%s' and status/state eq '%s' and status/reason eq '%s'",
+					*apiScanConfig.Id,
+					apitypes.AssetScanStatusStateDone,
+					apitypes.AssetScanStatusReasonSuccess,
+				)),
+			}
+			gomega.Eventually(func() bool {
+				scans, err = client.GetScans(ctx, scanParams)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				return len(*scans.Items) == 1
+			}, time.Second*120, time.Second).Should(gomega.BeTrue())
+
+			ginkgo.By("verifying that at least one package was found")
+			gomega.Eventually(func() bool {
+				totalPackages := (*scans.Items)[0].Summary.TotalPackages
+				return *totalPackages > 0
+			}, DefaultTimeout, time.Second).Should(gomega.BeTrue())
+		})
+	})
+
+	ginkgo.AfterEach(func(ctx ginkgo.SpecContext) {
+		if ginkgo.CurrentSpecReport().Failed() {
+			reportFailedConfig.startTime = ginkgo.CurrentSpecReport().StartTime
+			ReportFailed(ctx, testEnv, client, &reportFailedConfig)
+		}
+	})
+})
+
+// nolint:gomnd
+func createTestDir(dir string) (func(), error) {
+	err := os.MkdirAll(dir, 0o755)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create test dir: %w", err)
+	}
+
+	// Create a test file to print a UUID
+	testFile := filepath.Join(dir, "test.go")
+	testScript := "package main\n\nimport (\n\"fmt\"\n\"github.com/google/uuid\"\n)\n\nfunc main() {fmt.Println(uuid.New().String())}\n"
+	err = os.WriteFile(testFile, []byte(testScript), 0o600)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create test file: %w", err)
+	}
+
+	// Create a mod file to include the UUID package
+	testModFile := filepath.Join(dir, "go.mod")
+	testModScript := "module test\n\ngo 1.21.7\n\nrequire github.com/google/uuid v1.6.0\n"
+	err = os.WriteFile(testModFile, []byte(testModScript), 0o600)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create test mod file: %w", err)
+	}
```

Comment:
Instead of creating files during tests, the testing data could be added under `e2e/testdata/` directory (scripts, files, etc) and mounted to the scanner as a scan dir.

Reply:
great point 👍 let me change this